### PR TITLE
Add .whitesource and config files to activate whitesource integration

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description

We @bbarani already enable the access of [WhiteSource integration](https://whitesource.atlassian.net/wiki/spaces/WD/pages/697696422/WhiteSource+for+GitHub.com) with Github.com for this repo. However, the automatic PR of .whitesource is not created. We asked for the support from WhiteSource side and they suggested we could raise one by ourselves. This PR will also set the WhiteSource integration config mode `LOCAL` so it will be using the `whitesource.config` in the root directory. Dashboards team can modify this configuration on their own to customize it. We are providing the one we had for all repos at this time.

Another PR we created for the same issue. https://github.com/opensearch-project/OpenSearch-Dashboards/pull/999

Please be aware that when this PR is merged, WhiteSource integration might be automatically created CVEs Github issues like [these](https://github.com/opensearch-project/opensearch-build/issues?q=is%3Aopen+is%3Aissue+label%3A%22security+vulnerability%22) in build repo.

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [ ] New functionality includes testing.
  - [ ] All tests pass
- [x] New functionality has been documented.
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .whitesource [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "configExternalURL": "",
+    "projectToken": "",
+    "baseBranches": []
+  },
+  "checkRunSettings": {
+    "vulnerableCheckRunConclusionLevel": "failure",
+    "displayMode": "diff"
+  },
+  "issueSettings": {
+    "minSeverityLevel": "LOW"
```
Comment (ylwu-amzn) [inline .whitesource:13]: Do we have agreement on all repos that we should care `LOW` level issues?

File: whitesource.config [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+#log.files.maxFilesCount=
+#log.files.path=
+
+########################################
+# Package Manager Dependency resolvers #
+########################################
+resolveAllDependencies=false
+#excludeDependenciesFromNodes=.*commons-io.*,.*maven-model
+
+resolveAllDependencies=false
```
Comment (ylwu-amzn) [inline whitesource.config:81]: Duplicate line as line 78

Comment (zelinh) [reply whitesource.config:80]: Removed. Thanks!

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Comment by ohltyler** [comment]
@zelinh we have seen with Dependabot that oftentimes the alerts are for dependencies stored in `yarn.lock` that are actually coming from core Dashboards, but pulled into the plugin's dependency tree, and can't be fixed at the plugin level. Do you know if this will add similar issues, such as automatic PRs for dependencies stored in core Dashboards?

**Comment by zelinh** [comment]
> @zelinh we have seen with Dependabot that oftentimes the alerts are for dependencies stored in `yarn.lock` that are actually coming from core Dashboards, but pulled into the plugin's dependency tree, and can't be fixed at the plugin level. Do you know if this will add similar issues, such as automatic PRs for dependencies stored in core Dashboards?

I think this will mostly create issues for those vulnerabilities as long as they are existed in this repo. You can always whitelist those issues that can't fix at your level in the WhiteSource dashboard to ignore them. Furthermore, automatic fix for CVEs is disabled in default, so it won't create PR to fix CVEs unless you allow it.

**Comment by ohltyler** [comment]
> > @zelinh we have seen with Dependabot that oftentimes the alerts are for dependencies stored in `yarn.lock` that are actually coming from core Dashboards, but pulled into the plugin's dependency tree, and can't be fixed at the plugin level. Do you know if this will add similar issues, such as automatic PRs for dependencies stored in core Dashboards?
> 
> I think this will mostly create issues for those vulnerabilities as long as they are existed in this repo. You can always whitelist those issues that can't fix at your level in the WhiteSource dashboard to ignore them. Furthermore, automatic fix for CVEs is disabled in default, so it won't create PR to fix CVEs unless you allow it.

Got it, sounds good. Thanks for adding!

**Review by ohltyler — APPROVED** [review]

**Pull request merged**